src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,805 @@
+#
+########################################################################
+
+#=
+function intersect(V::T...) where T<:MatrixGroup
+   L = GAP.julia_to_gap([G.X for G in V])
+   K = GAP.Globals.Intersection(L)
+   return _as_subgroup(V[1], K)
+end
+
+function intersect(V::AbstractVector{T}) where T<:MatrixGroup
+   L = GAP.julia_to_gap([G.X for G in V])
+   K = GAP.Globals.Intersection(L)
+   return _as_subgroup(V[1], K)
+end
+=#
+
+########################################################################
+#
+# Conjugation
+#
+########################################################################
+
+function Base.show(io::IO, x::GroupConjClass{T,S}) where T <: MatrixGroup where S <: MatrixGroupElem
+  show(x.repr)
+  print(" ^ ")
+  show(x.X)
+end
+
+function Base.show(io::IO, x::GroupConjClass{T,S}) where T <: MatrixGroup where S <: MatrixGroup
+  show(x.repr)
+  print(" ^ ")
+  show(x.X)
+end
+
+function Base.:^(H::MatrixGroup, y::MatrixGroupElem)
+   if isdefined(H,:gens)
+      K = MatrixGroup(H.deg, H.ring)
+      K.gens = [y^-1*x*y for x in H.gens]
+      return K
+   else
+      return MatrixGroup(H.deg,H.ring,H.X^y.X)
+   end
+end
+
+function conjugacy_classes_subgroups(G::MatrixGroup)
+   L=GAP.gap_to_julia(Vector{GapObj},GAP.Globals.ConjugacyClassesSubgroups(G.X))
+   return GroupConjClass{typeof(G), typeof(G)}[ _conjugacy_class(G,MatrixGroup(G.deg,G.ring,GAP.Globals.Representative(cc)),cc) for cc in L]
+end
+
+function conjugacy_classes_maximal_subgroups(G::MatrixGroup)
+  L = GAP.gap_to_julia(Vector{GapObj},GAP.Globals.ConjugacyClassesMaximalSubgroups(G.X))
+   return GroupConjClass{typeof(G), typeof(G)}[ _conjugacy_class(G,MatrixGroup(G.deg,G.ring,GAP.Globals.Representative(cc)),cc) for cc in L]
+end
+
+function Base.rand(C::GroupConjClass{S,T}) where S<:MatrixGroup where T<:MatrixGroup
+   return MatrixGroup(C.X.deg,C.X.ring,GAP.Globals.Random(C.CC))
+end
+
+function elements(C::GroupConjClass{S, T}) where S where T<:MatrixGroup
+   L=GAP.Globals.AsList(C.CC)
+   l = Vector{T}(undef, length(L))
+   for i in 1:length(l)
+      l[i] = _as_subgroup(C.X, L[i])
+   end
+   return l
+end
+
+
+
+
+
+#=
+
+########################################################################
+#
+# conversions     (this should be no longer necessary, and it's also wrong)
+#
+########################################################################
+#TODO: support other field types
+
+function FieldGapToOscar(F::GapObj)
+   p = GAP.Globals.Characteristic(F)
+   q = GAP.Globals.Size(F)
+   d = Base.Integer(log(p,q))
+
+   return GF(p,d)
+end
+
+function FieldOscarToGap(F::FqNmodFiniteField)
+   p = Int64(characteristic(F))
+   d = Int64(degree(F))
+
+   return GAP.Globals.GF(p,d)
+end
+
+function FieldElemGapToOscar(x::FFE, Fgap::GapObj, F::FqNmodFiniteField)
+   q = GAP.Globals.Size(Fgap)
+   z = gen(F)
+   if GAP.Globals.IsZero(x)
+      return 0*z
+   else
+      d = Integer(GAP.Globals.LogFFE(x, GAP.Globals.Z(q)))
+      return z^d
+   end
+end
+
```

Comment:
This conversion assumes that the two fields `Fgap` and `F` are defined w.r.t. the same polynomial, and that this polynomial is primitive.
According to the documentation of `FlintFiniteField` (`Nemo.GF` is not documented), this function uses the Conway polynomial when this  is available, otherwise a random sparse irreducible polynomial. (If the polynomial is not primitive, `gen(F)` cannot be used like in this function.)
In order to make the conversion safe, one could throw an error when the two defining polynomials are different.

Reply:
Yes, in fact the actual method to convert GAP fields to Oscar fields and viceversa is the one used in the file `iso_oscar_gap.jl`. I think the lines from 717 to the end of the file `MatGrp.jl` can be discarded.